update readme (#1)
DevExpress-Examples/Blazor-dxdatagrid-inline-editing: Readme.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T985618)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# DataGrid for Blazor - How to implement the inline edit mode with the DisplayTemplate

![Data Grid - Inline edit form](images/dxdatsgrid-inline-editing.gif)

<!-- run online -->
**[[Run Online]](https://codecentral.devexpress.com/351812089/)**
<!-- run online end -->

> This example illustrates a possible implementation of the case when users can edit values in the inline edit row. 

Reviewer: ```suggestion
This example illustrates how to allow users to edit values in the inline edit row. 

```

The main idea of the solution is to create the [DisplayTemplate](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxDataGridColumn.DisplayTemplate) for each column and display the corresponding value or an editor based on the state of the current row. 

The **IsInEditMode** custom property defines the state of the current row. You can find the implementation of this custom property in the [WeatherForecastForEdit](./CS/BlazorGridInlineEditing/Data/WeatherForecastForEdit.cs) class. This class inherits the original *[WeatherForecast](./CS/BlazorGridInlineEditing/Data/WeatherForecast.cs)* model class and extends it with the **IsInEditMode** property. 

> Also the **WeatherForecastForEdit** class implements the [INotifyPropertyChanged](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged?view=net-5.0) interface. The Data Grid automatically detects such interfaces and handles the **PropertyChanged** event, so that if the corresponding property value changes, the DxDataGrid component re-renders the corresponding row. 

Reviewer: ```suggestion
The **WeatherForecastForEdit** class also implements the [INotifyPropertyChanged](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged?view=net-5.0) interface. The Data Grid automatically detects such interfaces and handles the **PropertyChanged** event to re-render the corresponding row if the corresponding property value changes. 

```
